### PR TITLE
Tighten warning for 'Requires.private' in simple pkg-config parser

### PR DESCRIPTION
**Pull request by jabraham17**
Tightens up the logic for when we should warn about 'Requires.private' (and also 'Requires'). We only need to warn when the compiler/linker flags are not already handled by 'Cflags'/`Libs`.

Details:

`Requires`/`Requires.private` list what other pkg-config libraries are needed, so that `pkg-config` can pull in the need dependencies/flags. This ends up being a recursive operation (i.e. 'A' has 'Requires: B', so we must process 'B.pc'). But if we already have the information needed, this is unnecessary. So if we have 'Requires: B', but 'B' is already listed in 'Libs', then for our purposes we have satisfied the 'Requires.'

Our simple pkg-config parser is used only for bundled third-party libs, and we do not need to ability to recursively search other .pc files to satisfy the requirements. So this PR uses the above logic to avoid warning about `Requires`/`Requires.private` not being handled

This should resolve https://github.com/chapel-lang/chapel/issues/26002

[Reviewed by @mppf  and @jhh67]

### CODE REVIEW

**Review by jhh67 — APPROVED** [review]
Very nice. Thanks for diagnosing the problem and coming up with a solution.

File: util/chplenv/third_party_utils.py [hunk 1]
```diff
@@ -236,11 +258,15 @@ def pkgconfig_get_bundled_link_args(pkg, ucp='', pcfile='',
...
+    if d.get('Requires') and _pkgconfig_should_warn_for_requires(d):
         warning("Simple pkg-config parser does not handle Requires")
         warning("in {0}".format(pcpath))
 
-    if static and 'Requires.private' in d and d['Requires.private']:
+    if (
+        static
+        and d.get("Requires.private")
+        and _pkgconfig_should_warn_for_requires(d, private=True)
+    ):
```
Comment (mppf) [inline util/chplenv/third_party_utils.py:269]: Trivial, but seems like really weird formatting. I would expect this:

```
if (static
    and d.get("Requires.private")
    and _pkgconfig_should_warn_for_requires(d, private=True)):
```

(But if there is some Python style convention that applies in this case, I wouldn't know about it).

**Review by mppf — APPROVED** [review]

File: util/chplenv/third_party_utils.py (hunk 1 above)
Comment (jabraham17) [reply util/chplenv/third_party_utils.py:269]: I used `black` to format the code, but I think both formattings match the PEP8 style guide

Comment (mppf) [reply util/chplenv/third_party_utils.py:269]: Alright! Thanks for checking.

**Pull request merged**